File: 71.简化路径.java
/*
 * @lc app=leetcode.cn id=71 lang=java
 *
 * [71] 简化路径
 */

// @lc code=start
import java.util.Stack;
class Solution {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack();
        String[] pathPart = path.split("/");
        for (String part: pathPart) {
            if (part.equals("") || part.equals(".")) {
                continue;
            }
            if (part.equals("..")) {
                if (!stack.empty()) {
                    stack.pop();
                }
            } else {
                stack.push(part);
            }
        }
        StringBuilder builder = new StringBuilder();
        builder.append("/");
        for (String part : stack) {
            builder.append(part);
            builder.append("/");
        }
        if (!stack.empty())
            builder.deleteCharAt(builder.lastIndexOf("/"));
        return builder.toString();
    }
}
// @lc code=end

File: 71.简化路径.py
#
# @lc app=leetcode.cn id=71 lang=python3
#
# [71] 简化路径
#

# @lc code=start


class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        path_part = path.split('/')
        for part in path_part:
            if part == '' or part == '.':
                continue
            if part == '..':
                if len(stack) > 0:
                    stack.pop()
            else:
                stack.append(part)
        return '/'+'/'.join(stack)
# @lc code=end
